enhancements/security/tls-artifacts-registry.md

```diff
@@ -0,0 +1,313 @@
+---
+title: tls-artifacts-registry
+authors:
+  - "@vrutkovs"
+  - "@dgrisonnet"
+  - "@dinhxuanvu"
+reviewers:
+  - "@deads2k"
+approvers:
+  - "@deads2k"
+api-approvers:
+  - "@deads2k"
+creation-date: 2023-10-26
+last-updated: 2023-12-08
+tracking-link:
+  - https://issues.redhat.com/browse/API-1603
+---
+
+# TLS Artifacts Registry
+
+## Summary
+
+This enhancement describes a single registry to store all produced 
+TLS artifacts. Both engineering and customers have expressed concern 
+about a lack of understanding what is the purpose of a particular 
+certificate or CA bundle, who issued it, and which team is responsible 
+for its contents and lifecycle. This enhancement describes a registry 
+to store necessary information about all certificates in the cluster 
+and sets requirements for TLS artifacts metadata.
+
+## Motivation
+
+TLS artifacts registry is useful for several teams in Red Hat (i.e. 
+which component manages a particular certificate) and customers - 
+for instance, helping them to separate platform-managed certificates 
+from user-managed.
+
+### User Stories
+
+- As an Openshift administrator, I want to differentiate 
+  platform-managed TLS artifacts from user-managed.
+- As an Openshift developer, I want to be able to find which team 
+  manages a particular certificate.
+- As an Openshift developer, I want to have a test which verifies 
+  that all certificates managed by the team have sufficient metadata
+  to explain its purpose
+- As a member of OpenShift concerned with the release process 
+  (TRT, dev, staff engineer, maybe even PM), I want to make sure all 
+  managed certificates have sufficient metadata to find the team 
+  responsible for its lifecycle.
+
+### Goals
+
+Create a registry of all managed TLS artifacts, and ensure that no new 
+certificates are added without sufficient metadata.
+Add a test to the conformance suite that verifies the actual cluster TLS 
+artifacts have necessary metadata.
+
+### Non-Goals
+
+* Include certificates provided by user (additional CA bundle, custom ingress / API certificates)
+* Include other encrypted artifacts - e.g. JWT tokens
```

Comment:
Maybe getting back to this in light of the recent discussions above. Is there any plan or idea on how this registry would be usable for the IBI/IBU flows or replacing recert if they are not including JWT or other derivatives? AFAIK @omertuc comments are basic requirements for replacement of recert.

Reply:
I'm not familiar with  IBI/IBU flows - do you have a link to describe those?

This registry won't replace recert, but hopefully it would be useful for recert tool - it would be a source of truth on cert locations and its properties, e.g. which signer has issued this cert and where on disk or in cluster this cert needs to be present etc.

Reply:
Regarding IBI/IBU - Theres [this high level write up](https://access.redhat.com/articles/7057298) which I recommend, but I think its irrelevant to the point im making.

I think this registry will not be that useful to recert [because recert will still have to iterate over all objects](https://github.com/rh-ecosystem-edge/recert/blob/95187641571861be024682bd5c4d0f697d38c503/src/cluster_crypto/crypto_objects.rs#L64-L90) for crypto analysis due to JWT explicitly being excluded from this EP. 

This is because before looking into the objects subject to modification, recert does not know where the JWTs / Certs are it has to modify. With this enhancement we could make recert use the certificate directory to skip one part of this, however JWTs would need either

- A) A new registry that is similar to the one discussed here
- B) An extension to this registry so that they can be mapped in this EP (removing it from non-goals)

Reply:
I think its worth extending this registry for JWT, although for this first phase its listed as a non-goal